lib/Logger.js

```diff
@@ -0,0 +1,86 @@
+let LogLevel = 0;
+
+const LogLevels = {
+    trace: -2,
+    debug: -1,
+    info: 0,
+    warn: 1,
+    error: 2
+};
+
+const BuildPrefix = (LogLevel) => {
+    let timestamp = new Date().toISOString();
+    return "[" + timestamp + "] [" + LogLevel + "] ";
+};
+
+class Logger {
+    /**
+     * @property {"trace"|"debug"|"info"|"warn"|"error"}
+     */
+    static get LogLevel() {
+        return Object.keys(LogLevels).find(key => LogLevels[key] === LogLevel);
+    }
+    static set LogLevel(value) {
+        if(!LogLevels[value]) {
+            throw "invalid log level '" + value + "'";
+        }
+        LogLevel = LogLevels[value];
+    }
+
+    /**
+     * @see console.trace
+     * @param {string} message 
+     * @param  {...any} args 
+     */
+    static trace(message, ...args) {
+        if(LogLevel <= LogLevels.trace) {
+            console.trace(BuildPrefix("TRACE") + message, ...args);
+        }
+    }
+
+    /**
+     * @see console.debug
+     * @param {string} message 
+     * @param  {...any} args 
+     */
+    static debug(message, ...args) {
+        if(LogLevel <= LogLevels.debug) {
+            console.debug(BuildPrefix("DEBUG") + message, ...args);
+        }
+    }
+
+    /**
+     * @see console.info
+     * @param {string} message 
+     * @param  {...any} args 
+     */
+    static info(message, ...args) {
+        if(LogLevel <= LogLevels.info) {
+            console.info(BuildPrefix("INFO") + message, ...args);
+        }
+    }
+
+    /**
+     * @see console.warn
+     * @param {string} message 
+     * @param  {...any} args 
+     */
+    static warn(message, ...args) {
+        if(LogLevel <= LogLevels.warn) {
+            console.warn(BuildPrefix("WARN") + message, ...args);
+        }
+    }
+
+    /**
+     * @see console.error
+     * @param {string} message 
```

Comment:
This is often used with an Error instance as the first argument.
Should the signature in general be just `...args` (i.e. without `message`)?

Reply:
It does make sense to have one defining message plus corresponding metadata. Otherwise this might get confusing.

I think we can enforce always adding a message to it